php/class-asset-manager-preload.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * Class file for Asset_Manager_Preload
+ *
+ * @package AssetManager
+ */
+
+/**
+ * Asset_Manager_Preload class.
+ */
+class Asset_Manager_Preload extends Asset_Manager {
+
+	/**
+	 * Holds references to the singleton instances.
+	 *
+	 * @var array
+	 */
+	private static $instance;
+
+	/**
+	 * Types of files that can be preloaded; corresponds to allowed `as` attribute values.
+	 *
+	 * @var array
+	 */
+	public $preload_as = [
+		'audio',
+		'document',
+		'embed',
+		'fetch',
+		'font',
+		'image',
+		'object',
+		'script',
+		'style',
+		'track',
+		'worker',
+		'video',
+	];
+
+	/**
+	 * Asset type this class is responsible for loading and managing.
+	 *
+	 * @var string
+	 */
+	public $asset_type = 'preload';
+
+
+	/**
+	 * Methods by which an asset can be loaded into the DOM.
+	 *
+	 * @var array
+	 */
+	public $load_methods = [ 'preload' ];
+
+	/**
+	 * Map of asset 'as' and 'type` attributes based on file extension, used to
+	 * patch in attributes for commonly-preloaded assets.
+	 * https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+	 *
+	 * @var array
+	 */
+	public $asset_types = [
+		'css'   => [
+			'as'        => 'style',
+			'mime_type' => 'text/css',
+		],
+		'js'    => [
+			'as'        => 'script',
+			'mime_type' => 'text/javascript',
+		],
+		'woff2' => [
+			'as'        => 'font',
+			'mime_type' => 'font/woff2',
+		],
+	];
+
+	/**
+	 * Constructor.
+	 */
+	private function __construct() {
+		// Don't do anything, needs to be initialized via instance() method.
+	}
+
+	/**
+	 * Get an instance of the class.
+	 *
+	 * @return class
```

Comment:
This phpdoc should be `@return Asset_Manager_Preload` ... An object is being returned and that kind specifically :)